<launch>

  <arg name="config" default="ASV_PPOcontinuous_MLP.yaml"/>
  <arg name="policy" default="ASV_GoThroughPosition.pth"/>
  <arg name="simulation" default="true"/>
  <arg name="cmd_topic" default="/rl_agent/cmd_drive"/>
  <arg name="odom_topic" default="/local_odom"/> <!-- /pose_gt has twist in world frame. Use test enviroment-->
  <arg name="goal_topic" default="/move_base_simple/goal"/>
  <arg name="status_topic" default="/agent_status"/>

  <node pkg="kingfisher_rl" type="capture_agent.py" name="rl_agent" output="screen">
    <remap from="~goal" to="$(arg goal_topic)"/>
    <remap from="~odom" to="$(arg odom_topic)"/> 
    <remap from="~rl_status" to="$(arg status_topic)"/>
    <remap from="~cmd_drive" to="$(arg cmd_topic)"/>

    <param name="config_file" value="$(arg config)"/>
    <param name="policy_file" value="$(arg policy)"/>
    <param name="dist_threshold" value="0.3"/>
    <param name="control_freq" value="20"/>
    <param name="obs_buffer_len" value="1"/>
  </node>

</launch>
